index.js

```diff
@@ -10,122 +10,122 @@ var url = require('url')
-  assert.equal(typeof cb, 'function', 'township-github._oauth: cb should be type Function')
+  ctx._verify = function (opts, cb) {
+    var code = opts.code
+    ctx._oauth(code, function (user) {
+      auth.db.get(opts.key, function (err, account) {
+        if (err) return cb(err)
+        cb(null, { key: account.key, github: { username: account.github.username } })
+      })
+    })
+  }
 
-  var self = this
+  ctx._oauth = function (code, cb) {
+    assert.equal(typeof code, 'string', 'township-github._oauth: code should be type String')
+    assert.equal(typeof cb, 'function', 'township-github._oauth: cb should be type Function')
 
-  var opts = xtend(this.verifyOpts, {
-    qs: {
-      client_secret: this.secret,
-      client_id: this.id,
-      code: code
-    }
-  })
+    var opts = xtend(ctx.verifyOpts, {
+      qs: {
+        client_secret: ctx.secret,
+        client_id: ctx.id,
+        code: code
+      }
+    })
 
-  var req = request(opts)
-  _parseBody(req, function (err, obj) {
-    if (err) return cb(err)
-    if (!obj) return cb(new Error('township-github._oauth: no response body received from GitHub'))
-    if (!obj.access_token) return cb(new Error('township-github._oauth: no access_token in body received from GitHub'))
+    var req = request(opts)
+    _parseBody(req, function (err, obj) {
+      if (err) return cb(err)
+      if (!obj) return cb(new Error('township-github._oauth: no response body received from GitHub'))
+      if (!obj.access_token) return cb(new Error('township-github._oauth: no access_token in body received from GitHub'))
 
-    var token = obj.access_token
+      var token = obj.access_token
 
-    self._getUser(token, cb)
-  })
-}
+      ctx._getUser(token, cb)
+    })
+  }
 
-Github.prototype._getUser = function (token, cb) {
-  assert.equal(typeof token, 'string', 'township-github._getUser: token.access_token should be type String')
-  assert.equal(typeof cb, 'function', 'township-github._getUser: cb should be type Function')
+  ctx._getUser = function (token, cb) {
+    assert.equal(typeof token, 'string', 'township-github._getUser: token.access_token should be type String')
+    assert.equal(typeof cb, 'function', 'township-github._getUser: cb should be type Function')
 
-  var opts = xtend(this.userOpts, {
-    qs: { access_token: token }
-  })
+    var opts = xtend(ctx.userOpts, {
+      qs: { access_token: token }
+    })
 
-  var req = request(opts)
-  pump(req, concat({ encoding: 'string' }, handler), function (err) {
-    if (err) return cb(explain(err, 'township-github._getUser: pipe error'))
-  })
+    var req = request(opts)
+    pump(req, concat({ encoding: 'string' }, handler), function (err) {
+      if (err) return cb(explain(err, 'township-github._getUser: pipe error'))
+    })
 
-  function handler (user) {
-    cb(null, user)
+    function handler (obj) {
+      var user = JSON.parse(obj)
+      cb(user)
+    }
   }
+
+  return ctx
```

Comment:
Now that we're moving off a prototype style onto a functional style I think it might be better to return a single object ASAP and rely on hoisting to expose stuff. E.g.

```js
function townshop () {
  var foo = bar

  return {
    create: create,
    login: login
  }

  function create () {}
  function login () {}
}
```